src/sentry/reprocessing2.py

```diff
@@ -347,7 +346,75 @@ def _get_info_reprocessed_key(group_id):
     return f"re2:info:{group_id}"
 
 
-def mark_event_reprocessed(data=None, group_id=None, project_id=None):
+def buffered_handle_remaining_events(
+    project_id,
+    old_group_id,
+    new_group_id,
+    datetime_to_event,
+    remaining_events,
+    force_flush_batch=False,
+):
+    """
+    A quick-and-dirty wrapper around `handle_remaining_events` that batches up
+    event IDs in Redis. We need this because Snuba cannot handle many tiny messages and prefers big ones instead.
+
+    Best performance is achieved when timestamps are close together. Luckily we
+    happen to iterate through events ordered by timestamp.
+
+    Ideally we'd have batching implemented via a service like buffers, but for
+    more than counters.
+    """
+
+    client = _get_sync_redis_client()
+    key = f"re2:remaining:{project_id}:{old_group_id}"
+
+    if datetime_to_event:
+        client.lpush(
+            key,
+            *(f"{to_timestamp(datetime)};{event_id}" for datetime, event_id in datetime_to_event),
+        )
+        client.expire(key, settings.SENTRY_REPROCESSING_SYNC_TTL)
+
+    if (
+        force_flush_batch
+        or client.llen(key) > settings.SENTRY_REPROCESSING_REMAINING_EVENTS_BUF_SIZE
+    ):
+        event_ids_batch = []
+        min_datetime = None
+        max_datetime = None
+
+        # TODO: Redis 6 introduces LPOP with <count> argument, use here?
+        while True:
+            row = client.lpop(key)
+            if row is None:
+                break
+
+            datetime_raw, event_id = row.split(";")
+            datetime = to_datetime(float(datetime_raw))
+
+            assert datetime is not None
+
+            if min_datetime is None or datetime < min_datetime:
+                min_datetime = datetime
+            if max_datetime is None or datetime > max_datetime:
+                max_datetime = datetime
+
+            event_ids_batch.append(event_id)
+
+        from sentry.tasks.reprocessing2 import handle_remaining_events
+
+        handle_remaining_events.delay(
+            project_id=project_id,
+            old_group_id=old_group_id,
+            new_group_id=new_group_id,
+            event_ids=event_ids_batch,
```

Comment:
This sends about 20kB into the task queue. Are we comfortable with such large argument lists?

Reply:
I thought we draw the line at event-payload-sized payloads, but I think it may be easy enough to refactor this code to simply keep things in Redis

Reply:
> but I think it may be easy enough to refactor this code to simply keep things in Redis

It would be a nice separation if one task pushes to redis and the other task reads from it (but that's a nit).